effekt/shared/src/main/scala/effekt/core/Transformer.scala

```diff
@@ -416,8 +416,8 @@ object Transformer extends Phase[Typechecked, CoreTransformed] {
           insertBindings { core.If(transformAsPure(cond), thenBranch, elseBranch) }
         case _ =>
           insertBindings {
-            val thenClause = preprocess(Nil, guards, thenBranch)
-            val elseClause = preprocess(Nil, Nil, elseBranch)
+            val thenClause = preprocess("guard_thn", Nil, guards, thenBranch)
+            val elseClause = preprocess("guard_els", Nil, Nil, elseBranch)
```

Comment:
maybe to be consistent with `whileThen`, do this instead?
```suggestion
            val thenClause = preprocess("guardThn", Nil, guards, thenBranch)
            val elseClause = preprocess("guardEls", Nil, Nil, elseBranch)
```